### DISCUSSION TITLE
Manipute query before it is handled by adaptor

### DISCUSSION THREAD

**Discussion by mayank-pathania** [betodealmeida/shillelagh#418] [Q&A]
Hi everyone,
Problem statement: take "WHERE" part of query as input and using that manipulate the "FROM" part of query (set as query params or headers in rest uri).

any approach or ideas for this will be helpful, because as far as i could explore, adaptors are not getting the "WHERE" part of query.

**Comment by betodealmeida** [comment]
Adapters receive the predicates in the `bounds` variable that is passed to `get_data`. Here's a minimal example:

```python
from shillelagh.adapters.registry import registry
from shillelagh.backends.apsw.db import connect

from fakes import FakeAdapter


class DebugAdapter(FakeAdapter):
    scheme = "debug"

    def get_data(self, bounds, *args, **kwargs):
        print(bounds)
        return super().get_data(bounds, *args, **kwargs)


registry.add("debug", DebugAdapter)

connection = connect(":memory:", ["debug"])
cursor = connection.cursor()
cursor.execute('SELECT * FROM "debug://" WHERE age = 1')
```

And running it:

```bash
% cd shillelagh/tests/
% python test.py
{'age': ==1}
```

**Reply by betodealmeida** [reply]
If you want the predicates during adapter instantiation (so you can have a dynamic table name), then that's not supported, since the adapter is not instantiated for every query.
